middleware/timeout_test.go

```diff
@@ -0,0 +1,90 @@
+// +build go1.13
+
+package middleware
+
+import (
+	"errors"
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestTimeout(t *testing.T) {
+	e := echo.New()
+
+	t.Run("Skipper", func(t *testing.T) {
+		t.Parallel()
+		m := TimeoutWithConfig(TimeoutConfig{
+			Skipper: func(context echo.Context) bool {
+				return true
+			},
+		})
+
+		req := httptest.NewRequest(http.MethodGet, "/", nil)
+		rec := httptest.NewRecorder()
+		c := e.NewContext(req, rec)
+
+		err := m(func(c echo.Context) error {
+			assert.NotEqual(t, "*context.timerCtx", reflect.TypeOf(c.Request().Context()).String())
+			return nil
+		})(c)
+
+		assert.NoError(t, err)
+	})
+
+	t.Run("Is cancelable", func(t *testing.T) {
+		t.Parallel()
+		m := TimeoutWithConfig(TimeoutConfig{
+			Timeout: time.Minute,
+		})
+
+		req := httptest.NewRequest(http.MethodGet, "/", nil)
+		rec := httptest.NewRecorder()
+		c := e.NewContext(req, rec)
+
+		err := m(func(c echo.Context) error {
+			assert.EqualValues(t, "*context.timerCtx", reflect.TypeOf(c.Request().Context()).String())
+			return nil
+		})(c)
+
+		assert.NoError(t, err)
+	})
+
+	t.Run("Times out after the predefined timeout", func(t *testing.T) {
+		t.Parallel()
+		m := TimeoutWithConfig(TimeoutConfig{
+			Timeout: time.Second,
```

Comment:
Please add a test with a Timeout equal to 0

Reply:
Will do